src/Interpreters/Cache/FileCache.cpp

```diff
@@ -1007,7 +1007,7 @@ void FileCache::freeSpaceRatioKeepingThreadFunc()
         limits_satisfied = main_priority->collectCandidatesForEviction(
             desired_size, desired_elements_num, keep_up_free_space_remove_batch, stat, eviction_candidates, lock);
 
-#ifdef ABORT_ON_LOGICAL_ERROR
+#ifdef DEBUG_OR_SANITIZER_BUILD
```

Comment:
I definitely like the changes, however checking the code it might be hard to decide when to use which. What I mean is usually now when `DEBUG_OR_SANITIZER_BUILD` is used, we carry out extra checks to enforce the rules that we think are exist. However Those extra checks usually using `chassert` which only abort when `ABORT_ON_LOGICAL_ERROR` is defined. So maybe let's just keep the new macro and get rid of `ABORT_ON_LOGICAL_ERROR`. What do you think?

Reply:
for me it was also hard to understand should i do the replacement or not in many places. so i'm both hands for carrying on only with the new one.